internal/bminventory/inventory.go

```diff
@@ -1142,14 +1159,20 @@ func (b *bareMetalInventory) GenerateClusterISOInternal(ctx context.Context, par
 	}
 
 	if imageExists {
-		if err = b.updateImageInfoPostUpload(ctx, cluster, clusterProxyHash, params.ImageCreateParams.ImageType); err != nil {
+		if err = b.updateImageInfoPostUpload(ctx, cluster, clusterProxyHash, params.ImageCreateParams.ImageType, false); err != nil {
 			return nil, common.NewApiError(http.StatusInternalServerError, err)
 		}
 
 		log.Infof("Re-used existing cluster <%s> image", params.ClusterID)
 		b.eventsHandler.AddEvent(ctx, params.ClusterID, nil, models.EventSeverityInfo, "Re-used existing image rather than generating a new one", time.Now())
 		return b.GetClusterInternal(ctx, installer.GetClusterParams{ClusterID: *cluster.ID})
 	}
+
+	// set image-generated indicator to false before the attempt to genearate the image, in order to have an explicit
+	// state of the image creation based on the cluster parameters which are already committed to the DB
+	if err = b.updateClusterImageGenerated(cluster, false); err != nil {
```

Comment:
maybe we can set updates[generated]=false at line 1138, and skip calling updateClusterImageGenerated here?

Reply:
correct, that can be simplified since in relies on the value of `imageExists`
`